### PR TITLE
[5631] Approve a pending publish request failed with a blank page

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5631

### CODE REVIEW

**Comment by jvega190** [comment]
Updated AwaitingForApproval and ApprovedScheduledItems dashlets to do what the RecentActivity dashlet does right now. RecentlyPublish doesn't need the update because there is no selection in that dashlet

File: ui/app/src/components/LegacySiteDashboard/LegacyApprovedScheduledDashlet/ApprovedScheduledDashlet.tsx [hunk 1]
```diff
@@ -98,6 +99,7 @@ export function ApprovedScheduledDashlet() {
     [isAllChecked, selectedLookup, state.itemsLookup]
   );
   const locale = useLocale();
+  const { itemsByPath } = useDetailedItems(Object.keys(selectedLookup));
```
Comment (rart) [inline ui/app/src/components/LegacySiteDashboard/LegacyApprovedScheduledDashlet/ApprovedScheduledDashlet.tsx]: Can we get by with SandboxItems here, in all these cases? Detailed items are more expensive.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/state/actions/content.ts [hunk 2]
```diff
@@ -48,6 +48,9 @@ export const fetchSandboxItemComplete = /*#__PURE__*/ createAction<{ item: Sandb
   'FETCH_SANDBOX_ITEM_COMPLETE'
 );
 export const fetchSandboxItemFailed = /*#__PURE__*/ createAction<AjaxError>('FETCH_SANDBOX_ITEM_FAILED');
+export const fetchSandboxItems = /*#__PURE__*/ createAction<{ paths: string[] }>('FETCH_SANDBOX_ITEMS');
+export const fetchSandboxItemsComplete = /*#__PURE__*/ createAction<SandboxItem[]>('FETCH_SANDBOX_ITEMS_COMPLETE');
```
Comment (rart) [inline ui/app/src/state/actions/content.ts]:
```suggestion
export const fetchSandboxItemsComplete = /*#__PURE__*/ createAction<{ items: SandboxItem[] }>('FETCH_SANDBOX_ITEMS_COMPLETE');
```

File: ui/app/src/state/epics/content.ts [hunk 3]
```diff
@@ -229,6 +233,19 @@ const content: CrafterCMSEpic[] = [
...
     ),
   // endregion
+  // region fetchSandboxItems
+  (action$, state$) =>
+    action$.pipe(
+      ofType(fetchSandboxItems.type),
+      withLatestFrom(state$),
+      switchMap(([{ payload }, state]) =>
+        fetchItemsByPath(state.sites.active, payload.paths).pipe(
+          map((items) => fetchSandboxItemsComplete(items as SandboxItem[])),
```
Comment (rart) [inline ui/app/src/state/epics/content.ts]:
```suggestion
          map((items) => fetchSandboxItemsComplete({ items })),
```

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/state/epics/content.ts [hunk 4]
```diff
@@ -98,7 +102,7 @@ import { AjaxError } from 'rxjs/ajax';
 import { showErrorDialog } from '../reducers/dialogs/error';
 import { dissociateTemplate } from '../actions/preview';
 import { isBlank } from '../../utils/string';
-import { DetailedItem } from '../../models';
+import { DetailedItem, SandboxItem } from '../../models';
```
Comment (rart) [inline ui/app/src/state/epics/content.ts]: Doesn't look like SandboxItem is in use. 

(Build & review for warnings always before final push)

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**